test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(NACKFRAG_COUNT_TOPIC));
+    // This topic uses the count_type that has been checked previously.
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            GAP_COUNT_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(GAP_COUNT_TOPIC));
+    // This topic uses the count_type that has been checked previously.
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            DATA_COUNT_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(DATA_COUNT_TOPIC));
+    // This topic uses the count_type that has been checked previously.
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            PDP_PACKETS_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PDP_PACKETS_TOPIC));
+    // This topic uses the count_type that has been checked previously.
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            EDP_PACKETS_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(EDP_PACKETS_TOPIC));
+    // This topic uses the count_type that has been checked previously.
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            DISCOVERY_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(DISCOVERY_TOPIC));
+    EXPECT_EQ(discovery_type, statistics_participant->find_type(discovery_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            SAMPLE_DATAS_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(SAMPLE_DATAS_TOPIC));
+    EXPECT_EQ(sample_identity_count_type, statistics_participant->find_type(
+            sample_identity_count_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            PHYSICAL_DATA_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PHYSICAL_DATA_TOPIC));
+    EXPECT_EQ(physical_data_type, statistics_participant->find_type(physical_data_type.get_type_name()));
+
+    // 5. Enable an enabled statistics DataWriter
```

Comment:
```suggestion
    // 5. Enable an already enabled statistics DataWriter
```

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3